src/engine/QueryPlanner.cpp

```diff
@@ -1768,6 +1794,41 @@ std::vector<QueryPlanner::SubtreePlan> QueryPlanner::createJoinCandidates(
   return candidates;
 }
 
+// _____________________________________________________________________________
+auto QueryPlanner::createSpatialJoin(
+    SubtreePlan a, SubtreePlan b,
+    const std::vector<std::array<ColumnIndex, 2>>& jcs)
+    -> std::optional<SubtreePlan> {
+  using enum QueryExecutionTree::OperationType;
+  const bool aIsSpatialJoin = a._qet->getType() == SPATIAL_JOIN;
+  const bool bIsSpatialJoin = b._qet->getType() == SPATIAL_JOIN;
+  
+  if (!(aIsSpatialJoin || bIsSpatialJoin)
+      || (aIsSpatialJoin && bIsSpatialJoin)) {
+    return std::nullopt;
+  }
+
+  SubtreePlan spatialSubtreePlan = aIsSpatialJoin ? a : b;
+  SubtreePlan otherSubtreePlan = aIsSpatialJoin ? b : a;
+
+  std::shared_ptr<Operation> op = spatialSubtreePlan._qet->getRootOperation();
+  SpatialJoin* spatialJoin = static_cast<SpatialJoin*>(op.get());
+
+  if (spatialJoin->isConstructed()) {
+    return std::nullopt;
+  }
+
+  ColumnIndex ind = aIsSpatialJoin ? jcs[0][1] : jcs[0][0];
```

Comment:
1. This code currently assumes, that there is exactly one pair of join columns. You have to check this if you assume this.

2. The case of two variables in common is actually useful and implementable (basically you then only have a distance filter on the result), but this doesn't have to be contained in this PR. Just write the assertion and add a `TODO: For two join columns we could add a simple filter operation here `.